lib/peripheral.dart

```diff
@@ -30,8 +30,30 @@ class Peripheral {
           timeout: timeout);
 
   Stream<PeripheralConnectionState> observeConnectionState(
-          {bool emitCurrentValue = false}) =>
-      _manager.observePeripheralConnectionState(identifier, emitCurrentValue);
+          {bool emitCurrentValue = false,
+          bool unsubscribeOnCancel = false}) {
+
+        var streamTransformer = StreamTransformer<PeripheralConnectionState, PeripheralConnectionState>.fromHandlers(
```

Comment:
Is it needed? Couldn't it be handled by 
```
return _manager.observePeripheralConnectionState(identifier, emitCurrentValue)
   .takeWhile(
       (state) { 
           if (completeOnDisconnect) 
               return state != PeripheralConnectionState.disconnected 
           else 
               return true; 
        })
```

Reply:
If the answer is no, then please extract this behaviour to InternalBleManager and just forward the argument from here to there.